src/store/auth/actions.ts

```diff
@@ -88,3 +112,64 @@ export const checkAuthentication = () => {
     }
   }
 }
+
+export const signup = (userOptions) => {
+  return async (dispatch) => {
+    try {
+      dispatch(signupRequest());
+
+      const resSignup = await xively.idm.authentication.createUser(userOptions);
```

Comment:
Error handling is broken here right now. Currently if an error is thrown on line 121 due to a bad password you will see [object Object] for the error. I did some debugging and the issue is coming from getJson in comm.ts. I think this function was written before we were using fetch and it now does not quite work with the fetch set up.

Whats going on here is line 24 of comm.ts ```javascript res = await fetch(url, options) ``` returns a object indicating the request failed (due to a bad password) with a field res.ok = false. line 30 sees that it is false and uses the following code to determine the error message:

```javascript
        reject(new Error(body && body.message ? body.message
          : 'Server sent error code ' + res.status + '\n\n' + JSON.stringify(body)));
```
body.message is an object, instantiating an error with an object leads to the error looking like "Error: [object Object]"

This leads to the following screenshot because it resolves to body.message, which is an object, not a string.
<img width="351" alt="screen shot 2018-03-02 at 11 18 09 am" src="https://user-images.githubusercontent.com/7120180/36910881-ef933b34-1e0f-11e8-91de-e375652e9aa2.png">

getJson needs to be refactored, but be careful doing this and please test it because it is used in essentially every IDM/blueprint API call. Also, consider that line 173 of auth/actions.ts is still expecting to read a "strength" value off of the error to determine if the password was bad. Perhaps throw a custom error object if body.message is an object, or perhaps stringify the object before using it to instantiate the error, i'll leave the solution up to you, let me know if you have questions.

<img width="1071" alt="screen shot 2018-03-02 at 11 55 05 am" src="https://user-images.githubusercontent.com/7120180/36911109-917f5478-1e10-11e8-8131-53fe20aad356.png">
<img width="709" alt="screen shot 2018-03-02 at 11 54 58 am" src="https://user-images.githubusercontent.com/7120180/36911110-918acb5a-1e10-11e8-8e48-96700ad22413.png">